test/cycles/basis.jl

```diff
@@ -0,0 +1,69 @@
+@testset "Cycle Basis" begin
+
+    function evaluate(x,y)
+        @test length(x) == length(y)
+        for i=1:length(x)
+            @test isempty(setdiff(x[i],y[i]))
+            @test isempty(setdiff(x[i],y[i]))
+        end
+    end
+
+    # No Edges
+    ex = Graph(1)
+    expected_cyclebasis = Array{Int64,1}[]
+    for g in testgraphs(ex)
+        ex_cyclebasis = cycle_basis(g)
```

Comment:
`ex_cyclebasis = @inferred cycle_basis(g)`

Reply:
Alright, is this also necessary for the other test cases?

Reply:
It's really just to confirm that the function is type-stable. I think it's sufficient to place it in one or two places.